index.js

```diff
@@ -37,17 +37,29 @@ function eyesWithout(fn) {
 function eyesWith(fn) {
   return function () {
     var windowSize = eyes.defaultWindowSize;
-    if (isPassedWindowSizeArgument(arguments)) {
-      windowSize = arguments[2];
+    var specVersion = '';
+
+    if (isPassedParameterArgument(arguments)) {
+      var params = arguments[2];
+      // width or height of 0 will make the params window size to be ignored
+      if (params.width && params.height) {
+        windowSize = {width: params.width, height: params.height};
+      }
+
+      if (params.version) {
+        specVersion = ' version: ' + params.version;
+      }
+
       delete arguments[2];
     }
+
     var spec = fn.apply(this, arguments);
     var hooked = spec.beforeAndAfterFns;
     spec.beforeAndAfterFns = function () {
       var result = hooked.apply(this, arguments);
       result.befores.unshift({fn: function (done) {
         eyesOpen = true;
-        eyes.open(browser, appName, 'eyes.it ' + spec.getFullName(), windowSize).then(done);
+        eyes.open(browser, appName, 'eyes.it ' + spec.getFullName() + specVersion, windowSize).then(done);
```

Comment:
don't do the logic here. call a function `buildSpecName(spec, params)`.
the `if(params.version)` should be there